pkg/support/dump_test.go

```diff
@@ -38,3 +138,111 @@ func TestParseFileName(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, ts.String(), "2021-08-17 09:47:58 +0000 UTC")
 }
+
+func TestTimestampFilter(t *testing.T) {
+	ts := timestampFilter("2m")
+	currentTime := time.Now()
+	require.NotNil(t, ts)
+	expectedTime := currentTime.Add(-2 * time.Minute)
+	assert.WithinDuration(t, expectedTime, *ts, 10*time.Second)
+}
+
+func TestWriteFile(t *testing.T) {
+	basedir := "dir1/dir2"
+	resource := "networkpolicies"
+	data := []byte("antct -oyaml get networkpolicies")
+
+	fs := afero.NewMemMapFs()
+	err := writeFile(fs, filepath.Join(basedir, resource), resource, data)
+	require.NoError(t, err)
+
+	fileContents, err := afero.ReadFile(fs, filepath.Join(basedir, resource))
+	require.NoError(t, err)
+	require.Equal(t, data, fileContents)
+}
+
+func TestWriteYAMLFile(t *testing.T) {
+	basedir := "dir1/dir2"
+	resource := "pod.yaml"
+	yamlData := `apiVersion: v1
+kind: Pod
+metadata:
+  name: my-pod`
+
+	fs := afero.NewMemMapFs()
+	err := writeYAMLFile(fs, filepath.Join(basedir, resource), resource, yamlData)
+	require.NoError(t, err)
+
+	fileContents, err := afero.ReadFile(fs, filepath.Join(basedir, resource))
+	require.NoError(t, err)
+
+	actualContents := string(fileContents)
+	actualContents = strings.TrimPrefix(actualContents, "|-")
+	actualContents = strings.TrimSpace(actualContents)
+
+	expectedContents := `apiVersion: v1
+  kind: Pod
+  metadata:
+    name: my-pod`
+
+	require.Equal(t, expectedContents, actualContents)
+}
+
+func TestDumpAntctlGet(t *testing.T) {
+	basedir := "dir1/dir2"
+	name := "agentinfo"
+
+	exe := new(testExecuter)
+	fakeCmd := exe.Command("antctl", "-oyaml", "get", name)
+	expectedOutput, _ := fakeCmd.CombinedOutput()
```

Comment:
Just use `expectedOutput := fmt.Sprintf("antctl -oyaml get %s", name)`. Creating a `fakeCmd` and calling `CombinedOutput` takes some value away from the test (and hurts readability IMO).

Same applies to other tests below.

Reply:
Addressed the comments but then will we be not using the Command function added by Harshitha?

Reply:
the test uses `testExec`. A `testExec` instance is used by `dumpAntctlGet`. The `Command` method will be invoked as expected when `dumpAntctlGet` executes. The `Command` method will output the command name + args. We are then validating that this is correct by matching it to `expectedOutput`. There is no reason to call `Command` directly to build `expectedOutput`. If you remove the `Command` method definition and the test still passes, then there is a problem...

Reply:
Okay understood, I checked after removing Command function...The tests don't work .